### DISCUSSION TITLE
Support Enum as workflow places

### DISCUSSION THREAD

**Discussion by stephanvierkant** [symfony/symfony#44203] [Q&A] [answered]
I've tried the new Enum in PHP 8.1 and I love that Symfony supports this new feature in the Form component.

I'm using [bpolaszek/doctrine-native-enums](https://github.com/bpolaszek/doctrine-native-enums) (until Doctrine supports Enum) and I'd love to have this support in the Workflow component as well. 

I've changed my status attribute from a string to an Enum field and now this doesn't work anymore:

`$workflow->can($entity, $transition->getName()`

I'd be great if Symfony would support Enum in the Workflow Component as well!

**Comment by stof** [comment]
This is indeed a good idea. I'm converting this into an issue to make it a feature request. Help is of course welcome to implement that in Symfony 6.1 (once we open the 6.1 branch, which should happen soon)

**Reply by stephanvierkant** [reply]
Thanks! And if someone has an idea how to achieve this, please let me know. I'd like to help, but don't know where to start.

I see there is an issue with `\Symfony\Component\Workflow\MarkingStore\MethodMarkingStore::getMarking` because `(string) $marking` doesn't work: an Enum has no `__toString()` (and can't even have it, anybody knows why?). I'm thinking of something like this:

```
if ($marking instanceof \BackedEnum) {
    $marking = $marking->value;
}
```

But before I'm digging into this, I'd like to hear some feedback before I'm spending time in creating stupid code.

**Comment by lyrixx** [comment, marked as answer]
Hello. I [answered](https://github.com/symfony/symfony/issues/44211) in the issue. I let you read the detail to understand why I'm "closing" this discussion as "won't" fix. 

Thanks for raising this discussion.
